modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/ArtifactDeployer.java

```diff
@@ -20,10 +20,21 @@
    *
    * The deployer executes the artifact installation phases until the artifact is deployed After this method call the Artifact
    * will be installed in the container and started.
+   *  @param artifact artifact to be deployed
+   *  @param startArtifact Whether the artifact should be started after initialisation
+   */
+  void deploy(final T artifact, boolean startArtifact);
```

Comment:
This will break backwards compatibility. Add at least a default. There are some classes in the maven plugin which implements this interface.

Reply:
As you noticed, the other cases depend on another interface. My mistake. This is internal.